File: javascript/pangram/pangram.js
var Pangram = function (stream){

    this.stream = stream;
    this.isPangram = function(){

        var char_arr = this.stream.replace(/[^A-Za-z]/g,'').toLowerCase().split(''),
            alphabet = 26;

        char_len = char_arr.unique().length;

        if (char_len === alphabet)
            return true;
        else
            return false;
    };
};

//Found a solution for unique arrays here. (Not my solution)
//http://jszen.com/best-way-to-get-unique-values-of-an-array-in-javascript.7.html

Array.prototype.unique = function()
{
    var n = {}, r = [];
    console.log(this);
    for(var i = 0; i < this.length; i++) 
    { 
        if (!n[this[i]]) 
        {
            n[this[i]] = true; 
            r.push(this[i]); 
        }
    }
    return r;
}

module.exports = Pangram;

File: ruby/pangram/pangram.rb
# coding: utf-8
class Pangram

  VERSION = 1
  def self.is_pangram?(str)

    if (str == "")
      return false
    end

    alphabet = 26
    str = str.downcase
    str = str.gsub(/[^a-z]/, '')
    str = str.split('')
    str = str.uniq!
    str_size = str.length
  
    str_size == alphabet ? true : false

  end
end
